In [4]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')
import numpy as np
import xlrd

Mounted at /content/drive


In [2]:
art = pd.read_excel('/content/drive/My Drive/project/seoul_contest/예술학원.xlsx')

In [3]:
art.columns
art.drop(['Unnamed: 3', 'Unnamed: 4','Unnamed: 6', 'Unnamed: 7','Unnamed: 9', 'Unnamed: 10'],axis=1,inplace=True)

In [4]:
art.dropna(inplace=True)
# art.to_csv('예술학원.csv',encoding='cp949')
art

행정구역 생활밀접업종 2018년 2분기 2019년 2분기 2020년 2분기
2     서울시 전체  서비스업/      7772      7727      7845
4        종로구  서비스업/       117       113       130
6      청운효자동  서비스업/        16        15        15
8        사직동  서비스업/        11        11        13
10       삼청동  서비스업/         0         0         1
..       ...    ...       ...       ...       ...
892     성내2동  서비스업/        15        16        16
894     성내3동  서비스업/        23        20        22
896       길동  서비스업/        42        41        40
898     둔촌1동  서비스업/         3         3         2
900     둔촌2동  서비스업/        21        21        20

[450 rows x 5 columns]

In [5]:
art['행정구역'] = art['행정구역'].str.strip()
art['생활밀접업종'] = art['생활밀접업종'].str[:-1]

In [6]:
gu_list = []
for i in art['행정구역']:
  if i[-1] =='구':
    gu_list.append(i.strip())   

In [7]:
art['행정구역'] = art['행정구역'].str.strip()
art['생활밀접업종'] = art['생활밀접업종'].str[:-1]

In [8]:
gubun = []
for i in art['행정구역']:
  if i in gu_list:
    gubun.append(i)
  else:
    gubun.append(np.nan)

In [9]:
# art.to_csv('예술학원.csv',encoding='utf-8')
len(gubun),len(art.index)
art['시군구']= gubun
art.fillna(method='ffill',inplace=True)

In [10]:
art['행정구역구분'] = gubun

In [11]:
# 특정값 포함하는 행 제거하기
# mask = art['행정구역'].isin(gu_list)) 
# art[~mask].head()

In [12]:
art_ = art[art['행정구역구분'].isnull()][1:]
art_.drop(['행정구역구분'],axis=1,inplace=True)
art_

행정구역 생활밀접업종  2018년 2분기  2019년 2분기  2020년 2분기  시군구
6    청운효자동    서비스         16         15         15  종로구
8      사직동    서비스         11         11         13  종로구
10     삼청동    서비스          0          0          1  종로구
12     부암동    서비스          8          6          8  종로구
14     평창동    서비스          5          4          4  종로구
..     ...    ...        ...        ...        ...  ...
892   성내2동    서비스         15         16         16  강동구
894   성내3동    서비스         23         20         22  강동구
896     길동    서비스         42         41         40  강동구
898   둔촌1동    서비스          3          3          2  강동구
900   둔촌2동    서비스         21         21         20  강동구

[424 rows x 6 columns]

In [13]:
art_['구동'] = art_['시군구']+art['행정구역']

In [14]:
art_.to_pickle('행정동별_예술학원.pkl')

In [2]:
workbook = xlrd.open_workbook('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx')

In [3]:
sheetlist=[]
for i in workbook.sheet_names():
  sheetlist.append(i)

In [4]:
len(workbook.sheet_names())

11

In [5]:
sheetlist

['동부', '서부', '남부', '북부', '중부', '강동송파', '강서양천', '강남서초', '동작관악', '성동광진', '성북강북']

In [7]:
a = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='동부')
b = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='서부')
c = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='남부')
d = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='북부')
e = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='중부')
f = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='강동송파')
g = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='강서양천')
h = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='강남서초')
i = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='동작관악')
j = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='성동광진')
k = pd.read_excel('/content/drive/My Drive/project/seoul_contest/서울시+학원+등록+교습비등+정보공개(2020.+4.+30.+기준).xlsx',sheet_name='성북강북')



한꺼번에 concat !

In [8]:
frames = [a,b,c,d,e,f,g,h,i,j,k]
result = pd.concat(frames)

In [9]:
df = result[result['교습과목(반)'].str.contains('초',na=False)]
# result_ele = result['교습과목(반)'].str.contains('초')
# df = result[result_ele]
df
# df.head()

학원명                               학원주소  ... 설립자-성명 강사수
0           예맥보습학원    서울특별시 중랑구 동일로151길 30  2층3층 (묵동)  ...    조소정   2
1           예맥보습학원    서울특별시 중랑구 동일로151길 30  2층3층 (묵동)  ...    조소정   2
6           세진보습학원      서울특별시 중랑구 공릉로 58  2층4층일부 (묵동)  ...    정찬교   3
7           세진보습학원      서울특별시 중랑구 공릉로 58  2층4층일부 (묵동)  ...    정찬교   3
8           세진보습학원      서울특별시 중랑구 공릉로 58  2층4층일부 (묵동)  ...    정찬교   3
...            ...                                ...  ...    ...  ..
8001     가온다실용음악학원  서울특별시 성북구 삼선교로10길 47   2층 (삼선동2가)  ...    조성현   2
8002     가온다실용음악학원  서울특별시 성북구 삼선교로10길 47   2층 (삼선동2가)  ...    조성현   2
8005     가온다실용음악학원  서울특별시 성북구 삼선교로10길 47   2층 (삼선동2가)  ...    조성현   2
8006     가온다실용음악학원  서울특별시 성북구 삼선교로10길 47   2층 (삼선동2가)  ...    조성현   2
8129  김영덕색소폰실용음악학원      서울특별시 강북구 한천로 1084 , 3층 (수유동)  ...    김영덕   1

[34689 rows x 22 columns]

In [10]:
df = df[df['교습계열']=='보통교과']
df.drop_duplicates(subset=['학원명'],inplace=True)

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [11]:
df.columns
new = df[['학원명', '학원주소', '분야구분', '교습계열', '교습과정', '교습과목(반)', '정원','총교습비']]
new.head(30)

학원명  ...      총교습비
0              예맥보습학원  ...  155000.0
6              세진보습학원  ...  150000.0
13           대찬하브루타학원  ...  200000.0
39          장안로드맵수학학원  ...  170000.0
50          더하이스트수학학원  ...  160000.0
75             미래사관학원  ...  270000.0
91               필립학원  ...   80000.0
112            가나보습학원  ...  265000.0
124             맨하탄학원  ...  182000.0
125            한영보습학원  ...  240000.0
128          YBM잉글루학원  ...  190000.0
136          수학엠플러스학원  ...  120000.0
150            대정보습학원  ...  230000.0
154            동서보습학원  ...  160000.0
160       백년대계제1관선화학원  ...  250000.0
169       교원더퍼스트캠퍼스학원  ...  120000.0
183              으뜸학원  ...  140000.0
188              수재학원  ...  160000.0
217  쎈수학러닝센터·3030영어학원  ...  180000.0
226            하나에듀학원  ...  150000.0
239            면일보습학원  ...  150000.0
244            정상보습학원  ...  150000.0
261            표준보습학원  ...  170000.0
283            이문보습학원  ...  280000.0
292           중앙플러스학원  ...  170000.0
307            교우보습학원  ...  170000.0
308             뉴비전학원  ...  380000.0
316            일류보습학원  ...  150000.0
348     에프엠(FM)최강수학학원  ...  140000.0
371      에듀아이영어수학전문학원  ...  180000.0

[30 rows x 8 columns]

In [12]:
new.to_pickle('보습학원주소.pkl')

In [54]:
# df['학원주소'] = df['학원주소'][6:10]

In [13]:
gu = []
for i in df['학원주소']:
  gu.append(i[6:10].strip())
df['시군구명'] = gu

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.


In [57]:
df['행정동'] = dong

In [58]:
df.to_excel('학원.xlsx',encoding='cp949')

In [59]:
df

학원명  ...          행정동
0           예맥보습학원  ...         [묵동]
6           세진보습학원  ...         [묵동]
13        대찬하브루타학원  ...        [전농동]
39       장안로드맵수학학원  ...        [장안동]
50       더하이스트수학학원  ...        [전농동]
...            ...  ...          ...
7588      헤럴드스쿨어학원  ...        [종암동]
7606       청담강북어학원  ...  [미아동,미아프라자]
7619  EIE(이아이이)어학원  ...        [미아동]
7660     성북메가스터디학원  ...     [동소문동6가]
7684    하이스트미래탐구학원  ...        [돈암동]

[4989 rows x 10 columns]

In [3]:
import requests
from urllib.parse import urlparse
import geopandas as gpd
from tqdm.notebook import tqdm

In [2]:
!pip install geopandas

     |████████████████████████████████| 972kB 4.7MB/s 
     |████████████████████████████████| 10.9MB 14.5MB/s 
     |████████████████████████████████| 14.8MB 180kB/s 


In [71]:
add =[]
[add.append(i) for i  in tqdm(new['학원주소'])]
len(add)

4989

In [236]:
import re
new

학원명  ...      총교습비
0           예맥보습학원  ...  155000.0
6           세진보습학원  ...  150000.0
13        대찬하브루타학원  ...  200000.0
39       장안로드맵수학학원  ...  170000.0
50       더하이스트수학학원  ...  160000.0
...            ...  ...       ...
7588      헤럴드스쿨어학원  ...  120000.0
7606       청담강북어학원  ...  200000.0
7619  EIE(이아이이)어학원  ...  120000.0
7660     성북메가스터디학원  ...  340000.0
7684    하이스트미래탐구학원  ...   42500.0

[4989 rows x 8 columns]

In [244]:
new.to_excel('학우너주소.xlsx',encoding='cp949')

주소 전처리

In [237]:
pre_= []
for i in tqdm(new['학원주소']):
  pre = re.sub(r'\([^)]*\)', '', i)
  pre_.append(pre.split(',')[0])

In [238]:
len(pre_)

4989

In [173]:
url = "https://dapi.kakao.com/v2/local/search/address.json?&query=" + '서울 강서구 강서로47길'
result = requests.get(urlparse(url).geturl(),
                      headers={"Authorization":"KakaoAK 4d25f7ae6c4cab6bc96c746280e403af"})
json_obj = result.json()
print(json_obj)

{'documents': [{'address': None, 'address_name': '서울 강서구 강서로47길', 'address_type': 'ROAD', 'road_address': {'address_name': '서울 강서구 강서로47길', 'building_name': '', 'main_building_no': '', 'region_1depth_name': '서울', 'region_2depth_name': '강서구', 'region_3depth_name': '내발산동', 'road_name': '강서로47길', 'sub_building_no': '', 'underground_yn': 'N', 'x': '126.831239727011', 'y': '37.5531202090817', 'zone_no': ''}, 'x': '126.831239727011', 'y': '37.5531202090817'}], 'meta': {'is_end': True, 'pageable_count': 1, 'total_count': 1}}


In [247]:
for i in pre_:
  print(i)

서울특별시 중랑구 동일로151길 30  2층3층 
서울특별시 중랑구 공릉로 58  2층4층일부 
서울특별시 동대문구 전농로 120 
서울특별시 동대문구 한천로 124 
서울특별시 동대문구 사가정로 190  전농우성아파트 상가4층7
서울특별시 동대문구 천호대로16길 12  3층4층 
서울특별시 중랑구 동일로129길 54  2층 
서울특별시 중랑구 동일로130길 57   
서울특별시 동대문구 황물로 42 
서울특별시 동대문구 답십리로23길 59   
서울특별시 중랑구 면목로48길 49 
서울특별시 동대문구 한천로 165 
서울특별시 중랑구 용마산로 389 
서울특별시 중랑구 상봉중앙로 51 2층
서울특별시 중랑구 면목로 306  2층3층일부 
서울특별시 동대문구 이문로 65 
서울특별시 중랑구 용마산로 436 
서울특별시 동대문구 답십리로 133 
서울특별시 중랑구 면목로27길 6   
서울특별시 중랑구 봉우재로 211  2층 
서울특별시 중랑구 용마산로94길 47  2층) 
서울특별시 중랑구 면목로48길 65 
서울특별시 동대문구 제기로38길 50   
서울특별시 동대문구 한천로 481 
서울특별시 동대문구 천호대로77길 94  2층 
서울특별시 동대문구 사가정로 93 
서울특별시 중랑구 신내로7가길 19  3층
서울특별시 중랑구 동일로140길 15   
서울특별시 동대문구 답십리로 224-1  2층3층4층 
서울특별시 중랑구 면목로 376 
서울특별시 중랑구 동일로 814  2층일부 
서울특별시 동대문구 경희대로4길 79 
서울특별시 동대문구 답십리로69길 8 
서울특별시 동대문구 사가정로 148 
서울특별시 중랑구 용마산로116길 22  4층 
서울특별시 동대문구 무학로42길 45 
서울특별시 동대문구 한천로 97 2층3층
서울특별시 동대문구 약령시로 16 
서울특별시 중랑구 면목로31길 26   
서울특별시 동대문구 한천로46가길 6 
서울특별시 중랑구 망우로 505 
서울특별시 동대문구 장한로28길 48 
서울특별시 동대문구 답십리로72길 118  4층 

In [226]:
dong_list = []
for i in tqdm(pre_ad):
  url = "https://dapi.kakao.com/v2/local/search/address.json?&query=" + i
  result = requests.get(urlparse(url).geturl(),
                        headers={"Authorization":"KakaoAK 4d25f7ae6c4cab6bc96c746280e403af"})
  json_obj = result.json()
  for document in json_obj['documents']:
    try:
      dong_list.append(document['address']['region_3depth_h_name'])
    except :
      dong_list.append('0')
df = pd.DataFrame(dong_list, columns = ['address_name'])
df

address_name
0             묵2동
1             묵1동
2            전농2동
3            장안1동
4            전농2동
...           ...
4883         돈암1동
4884          종암동
4885         삼각산동
4886          미아동
4887          동선동

[4888 rows x 1 columns]

4989개를 돌렸는데 4887개만 return..\
Nonetype Error 는 try-except문으로 0 반환이 되었는데\
또 다른 오류가 있는 듯 ..하...\
카카오 api 왜이렇게 허술한거니..\

In [227]:
len(pre_ad)

4989

In [230]:
df.to_csv('동.csv',encoding='cp949')

In [235]:
len(pre_ad)

4989

test

In [233]:
new_list = []
for i in tqdm(pre_ad):
  url = "https://dapi.kakao.com/v2/local/search/address.json?&query=" + i
  result = requests.get(urlparse(url).geturl(),
                        headers={"Authorization":"KakaoAK 4d25f7ae6c4cab6bc96c746280e403af"})
  json_obj = result.json()
  for document in json_obj['documents']:
    try:
      l_list.append(document['address']['region_3depth_h_name'])
    except :
      l_list.append('0')
df2 = pd.DataFrame(l_list, columns = ['address_name'])
df2

address_name
0          묵2동
1          묵1동
2         전농2동
3         장안1동
4         전농2동

In [204]:
# dong_list = []
# def address_to_latlon_test(list):
dong_list = []
for i in pre_ad[26]:
  url = "https://dapi.kakao.com/v2/local/search/address.json?&query=" + i
  result = requests.get(urlparse(url).geturl(),
                        headers={"Authorization":"KakaoAK 4d25f7ae6c4cab6bc96c746280e403af"})
  json_obj = result.json()
  for document in json_obj['documents']:
    # dong_list.append(document['address']['region_3depth_h_name'])
    val = document['address']['region_3depth_h_name']
    # except :dong_list.append('0')
df = pd.DataFrame(val, columns = ['address_name'])
df

address_name
0          제기동

In [214]:
url = "https://dapi.kakao.com/v2/local/search/address.json?&query=" + '서울특별시 중랑구 용마산로 248  301호3층일부'
result = requests.get(urlparse(url).geturl(),
                      headers={"Authorization":"KakaoAK 4d25f7ae6c4cab6bc96c746280e403af"})
json_obj = result.json()
for document in json_obj['documents']:
  try:
    val = document['address']['region_3depth_h_name']
  except :
    val = ['0']
df = pd.DataFrame(val, columns = ['address_name'])
df

address_name
0            0

In [197]:
aa = address_to_latlon_test(tqdm(pre_ad[66]))#2866
aa

Empty DataFrame
Columns: [address_name]
Index: []

In [87]:
a = address_to_latlon(tqdm(pre_ad[:2865]))#2866

In [250]:
len(a)

2812

In [252]:
new['pre_address'] = pre_ad

In [256]:
foru= new[['학원명','학원주소','pre_address']]
foru.to_excel('초등보습학원주소.xlsx',)